api/v1/settings.go

```diff
@@ -12,25 +14,56 @@ type Settings struct {
 	MaxInFlightGiB uint `json:"max_in_flight_gib"`
 }
 
-// handleGetSettings godoc
-//
-//		@Summary		Apply new Tenant Settings
-//		@Security apiKey
-//	  @Param 			settings body Settings true "New settings to apply"
-//		@Produce		json
-//		@Success		200	{object}	ResponseEnvelope{response=Settings}
-//		@Router			/settings [post]
-func handleGetSettings(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) ConfigureSettingsRouter(e *echo.Group) {
+	g := e.Group("/settings")
+	g.POST("", a.handleSetSettings)
+	g.GET("", a.handleGetSettings)
 }
 
 // handleSetSettings godoc
 //
 //	@Summary		Get the currently active Tenant Settings
-//	@Security apiKey
+//	@Security		apiKey
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=Settings}
 //	@Router			/settings [get]
-func handleSetSettings(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) handleGetSettings(c echo.Context) error {
+	var tenant db.Tenant
+	tenant.TenantID = db.ID(GetTenantContext(c).TenantID)
+	res := a.db.Model(&db.Tenant{}).Find(&tenant)
+
+	if res.Error != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, res.Error.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, tenant.TenantSettings))
+}
+
+// handleGetSettings godoc
+//
+//	@Summary		Apply new Tenant Settings
+//	@Security		apiKey
+//	@Param			settings body Settings true "New settings to apply"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=Settings}
+//	@Router			/settings [post]
+func (a *apiV1) handleSetSettings(c echo.Context) error {
+	var settings Settings
+
+	err := json.NewDecoder(c.Request().Body).Decode(&settings)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	var tenant db.Tenant
+	tenant.TenantID = db.ID(GetTenantContext(c).TenantID)
+
+	res := a.db.Model(&tenant).UpdateColumn("tenant_settings", &settings)
```

Comment:
only thing to note is that POSTing settings here will overwrite the existing one completely, so if there are any ommitted/null fields I think it will reset them. Important to make a note of maybe in the godoc comment for now